libscript/src/objc.lcb

```diff
@@ -610,4 +610,89 @@ public handler CreateObjcInformalDelegateWithContext(in pProtocol as List, in pH
     end unsafe
 end handler    
 
+/* Block handling */
+
+type LibraryHandle is Pointer
+
+private variable mNSConcreteGlobalBlock as optional Pointer
+
+constant kRTLD_LAZY is 1
+private foreign handler OpenLibrary(in pPath as ZStringUTF8, in pFlag as CSInt) returns optional LibraryHandle binds to "dlopen"
+private foreign handler GetSymbolPointer(in pHandle as LibraryHandle, in pSymbol as ZStringUTF8) returns optional Pointer binds to "dlsym"
+private foreign handler CloseLibrary(in pHandle as LibraryHandle) returns CSInt binds to "dlclose"
+
+private unsafe handler EnsureNSConcreteGlobalBlock() returns nothing
+	if mNSConcreteGlobalBlock is not nothing then
+		return
+	end if
+
+	variable tSystem as LibraryHandle
+	put OpenLibrary("libSystem.dylib", kRTLD_LAZY) into tSystem
+	if tSystem is not nothing then
+		put GetSymbolPointer(tSystem, "_NSConcreteGlobalBlock") into mNSConcreteGlobalBlock
+		CloseLibrary(tSystem)
+	end if
+end handler
+
+public foreign type BlockDescriptor binds to "MCAggregateTypeInfo:ffrrr"
+public foreign type Block binds to "MCAggregateTypeInfo:rEErr"
+type BlockPointer is Pointer
+
+foreign handler MCHandlerGetFunctionPtr(in pHandler as any, out rPtr as Pointer) returns CBool binds to "<builtin>"
+
+private foreign handler CopyBlockDescriptor(inout pBlock as BlockDescriptor, in pSize as UIntSize, out rNewBlock as Pointer) returns CBool binds to "MCMemoryAllocateCopy"
+
+private foreign handler CopyBlock(inout pBlock as Block, in pSize as UIntSize, out rNewBlock as Pointer) returns CBool binds to "MCMemoryAllocateCopy"
+
+private foreign handler MCMemoryDelete(in pBlock as Pointer) returns nothing binds to "<builtin>"
+
+private foreign handler BlockPointerToBlock(out rBlock as Block, \
```

Comment:
We haven't continued args elsewhere, so retract that and just continue the `binds`